### ISSUE TITLE
[Bug]: cuDNN 9.26 SDPA backward plan creation segfaults when B=Hq=Sq=1

### ISSUE THREAD

**Issue by felixffl** [NVIDIA/cudnn-frontend#1071]
### Before submitting

- [x] I searched existing issues and did not find a duplicate.

### Affected area

None

### Description

cuDNN Backend 9.26.0.51 segfaults while creating an SDPA backward execution plan on H200 when `B=Hq=Hkv=Sq=1`, `Sk=128`, and `D=64` (BF16, non-causal).

The reproducer below uses cuDNN Frontend directly, without Transformer Engine or training data. The crash occurs in `create_execution_plans([cudnn.heur_mode.A])`, before any attention forward/backward execution. No negative attention scores are required.

I searched existing reports. [#628](https://github.com/NVIDIA/cudnn-frontend/pull/628) concerns a GPU backward negative-LSE/KV-tail NaN bug; this report is specifically about a host-side planning crash. I did not find an exact match for this crash.

### Steps to reproduce

1. Use cuDNN Frontend 1.26.0 with Backend 9.26.0.51 on H200. See the environment details below; the tested backend is the CUDA 12 wheel, with an existing CUDA 13 PyTorch installation.
2. Save the following as `repro.py` and run `python repro.py`.
3. Confirm that the printed backend version is `92600`.

```python
import torch
import cudnn

print(f"PyTorch: {torch.__version__}", flush=True)
print(f"Frontend: {cudnn.__version__}; Backend: {cudnn.backend_version()}", flush=True)
print(f"GPU: {torch.cuda.get_device_name()}", flush=True)
torch.manual_seed(123)
B, H, SQ, SK, D = 1, 1, 1, 128, 64

# No forward/backward execution is needed: the crash occurs during planning.
graph = cudnn.pygraph(
    io_data_type=cudnn.data_type.BFLOAT16,
    intermediate_data_type=cudnn.data_type.FLOAT,
    compute_data_type=cudnn.data_type.FLOAT,
)
values = {
    "q": torch.randn(B, H, SQ, D, device="cuda", dtype=torch.bfloat16),
    "k": torch.randn(B, H, SK, D, device="cuda", dtype=torch.bfloat16),
    "v": torch.randn(B, H, SK, D, device="cuda", dtype=torch.bfloat16),
    "o": torch.zeros(B, H, SQ, D, device="cuda", dtype=torch.bfloat16),
    "dO": torch.randn(B, H, SQ, D, device="cuda", dtype=torch.bfloat16),
    "stats": torch.zeros(B, H, SQ, 1, device="cuda", dtype=torch.float32),
}
tensors = {name: graph.tensor_like(value) for name, value in values.items()}
grads = graph.sdpa_backward(
    **tensors, attn_scale=D**-0.5, use_causal_mask=False,
)
for grad, name in zip(grads, ("q", "k", "v")):
    grad.set_output(True).set_dim(values[name].shape).set_stride(values[name].stride())
graph.validate()
graph.build_operation_graph()
print("before create_execution_plans", flush=True)
graph.create_execution_plans([cudnn.heur_mode.A])
print("after create_execution_plans", flush=True)
```

`o` and `stats` have finite placeholder values because this reproducer only builds a plan; it does not execute the graph or check gradients.

With the same reproducer and Frontend, replacing only the isolated Backend wheel with 9.25.1.1 prints `after create_execution_plans` and exits normally. The cuBLAS/NVRTC dependencies are held constant.

Earlier fresh-forward/backward controls with the same `Sk=128,D=64` showed that changing B to 2, Hq/Hkv to 2/2, or Sq to 2 avoids this crash. Avoiding the crash is not a claim that every such case has correct gradients; gradient NaNs are a separate issue.

### Expected behavior

Plan creation should succeed for this shape, or report a supported cuDNN/Frontend error if the configuration is unsupported. It should not terminate the process with SIGSEGV.

### Actual behavior

The process receives SIGSEGV immediately after `before create_execution_plans`. The subprocess return code is `-11`; `after create_execution_plans` is never printed.

GDB stops inside `libcudnn_engines_runtime_compiled.so.9`. The stack reaches it through `cudnn::cnn::EngineInterface::isSupported()` and backward engine heuristic selection. At the faulting instruction, `RDI=0x430`, so reading `0x10(%rdi)` attempts to access address `0x440`.

This is consistent with a null-base member access. I have not identified the internal object type or the source-level reason it is null, and am not attributing it to LSE broadcasting without further evidence.

### Environment (where applicable)

The environment summary below was verified using the official standalone `collect_env.py`. The full privacy-redacted report is available upon request. The installed Frontend 1.26.0 does not provide `python -m cudnn.collect_env`, so I used the standalone script linked in this issue template.

- cuDNN Frontend: **1.26.0**
- cuDNN Backend actually loaded: **9.26.0.51**, `cudnn.backend_version() == 92600`
- Backend wheel: **nvidia-cudnn-cu12**
- GPU: **NVIDIA H200, SM90**
- NVIDIA driver: **590.48.01**
- OS: **Ubuntu 24.04.4 LTS**, container; host kernel **5.15.0-119-generic**
- Python: **3.12.3**
- GCC: **13.3.0**
- PyTorch: **2.11.0+cu130**, `torch.version.cuda == "13.0"`
- CUDA toolkit on PATH: **13.0**, `nvcc V13.0.88`
- Isolated cuBLAS CUDA 12 dependency: **12.9.2.10**
- Isolated NVRTC CUDA 12 dependency: **12.9.86**

This is an intentional per-process Backend substitution, not a clean all-CUDA-12 environment: PyTorch remains the CUDA 13 build, while the cuDNN CUDA 12 wheel and its CUDA 12 dependencies are isolated through the library search path and preload. The collector reports both CUDA 12 and CUDA 13 cuBLAS libraries in the process; both library versions are disclosed here for completeness. Only the selected 9.26 cuDNN libraries are used for the failing Backend path. Other cuDNN installations remain on disk. The 9.25.1 comparison uses the same setup except for the Backend wheel.

Any logs shared on request will have personal paths, device UUIDs, and machine topology redacted while preserving library names and versions.

### Last known working version

**cuDNN Backend 9.25.1.1**, with the same Frontend 1.26.0 and the exact planning-only reproducer above.

| Backend | Planning-only reproducer |
|---|---|
| 9.25.1.1 (`92501`) | Prints both markers; exit code 0 |
| 9.26.0.51 (`92600`) | SIGSEGV before the second marker |

Here “working” means plan creation does not crash; this reproducer does not validate execution or numerical correctness.

### Relevant logs

```text
PyTorch: 2.11.0+cu130
Frontend: 1.26.0; Backend: 92600
GPU: NVIDIA H200
before create_execution_plans
```

GDB excerpt (the full stack is available upon request):

```text
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
#0  ... in ?? () from libcudnn_engines_runtime_compiled.so.9
...
#12 ... in cudnn::cnn::EngineInterface::isSupported()
#13 ... in cudnn::backend::EngineConfig::finalize_internal()
...
rbx  0x0
rdi  0x430
=> mov 0x10(%rdi),%rdx
```

Full environment and debug logs are available upon request. Frontend and Backend logs were captured with:

```shell
export CUDNN_FRONTEND_LOG_FILE=fe.log
export CUDNN_FRONTEND_LOG_INFO=1
export CUDNN_LOGLEVEL_DBG=3
export CUDNN_LOGDEST_DBG=be.log
python repro.py
```
